src/tool_operate.c

```diff
@@ -697,45 +697,59 @@ static CURLcode post_per_transfer(struct GlobalConfig *global,
   return result;
 }
 
+/* helper function to ensure input ends in val_to_ensure */
+static void ensure_trailing(char **input, const char val_to_ensure)
+{
+  if(*input && **input) {
+    size_t len = strlen(*input);
+    if(((*input)[len - 1] != val_to_ensure)) {
+      char *composed_input = aprintf("%s%c", *input, val_to_ensure);
+      if(composed_input) {
+        Curl_safefree(*input);
+        *input = aprintf("%s", composed_input);
```

Comment:
Save a useless copy:

```suggestion
        *input = composed_input;
```

Reply:
Fixed by restructuring it with the idea below from Stebalien.